docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Deployment as code
+description: You can define your entire Octopus configuration as code and track it all through source control. This page describes the different options available right now, and our plans to make this even better in the future.
+position: 4
+---
+
+You are probably reading this page because you want to manage your **deployment as code**. You want to define your deployment process, scripts, and variables in code, perhaps colocating this with the application source code in your git repository. This page describes the different options available in Octopus right now, and our plans to make this even better in the future.
```

Comment:
The opening paragraph is good as it is, but... one of the things I've been trying to do with the docs is make them more declarative and reduce any unnecessary words/phrases. We're getting more users who have English as a second language and find brevity easier to understand.
It also means it's easier for people to get to the heart of what we're saying as quickly as possible (by reducing the extra words we're hopefully training users not to skim).
I'd also prefer we avoid making future promises in the docs, often things change, but we don't remember to go back and update the doc where we've made future promises. The blog feels like a better place for that more speculative future focused content as it doesn't matter so much if blog posts become stale.
With that in mind, I'd change the opening paragraph to something like:
With Octopus you can manage your **deployments as code**. This means you can define your deployment process, scripts, and variables in code, perhaps colocating them with the application source code in your git repository. This page describes the different options available in Octopus to store your **deployments as code**.